x-pack/plugins/uptime/server/lib/helper/get_filtered_query_and_status.ts

```diff
@@ -169,9 +169,8 @@ export const getFilteredQueryAndStatusFilter = (
       bool: {
         must: filterObject.bool.must.filter(
           (filter: any) =>
-            (filter.match && !filter.match['monitor.status']) ||
-            filter.simple_query_string ||
-            filter.bool
+            // drop monitor.status filter clause
+            !get<any | undefined>(filter, ['match', 'monitor.status'], undefined)
```

Comment:
@justinkambic just curious why the arguments to `get` are in this array like this? Is this the same as
```ts
!get<any | undefined>(filter, 'match.monitor.status', undefined)
```

Out of scope here but this is a place where using elastic-idx could be nice, esp if `filter` was typed:
```ts
!idx(filter, _ => _.match.monitor.status)
```

Reply:
I could be wrong but I think `lodash` will interpret this as checking for:

```JSON
{
  "match": {
    "monitor": {
      "status": "someVal"
    }
  }
}
```

When in reality the object we receive is going to look like:

```JSON
{
  "match": {
    "monitor.status": "someVal"
  }
}
```

Reply:
> Out of scope here but this is a place where using elastic-idx could be nice, esp if filter was typed

Out of scope reply but could you link to an example or doc? If that's the case there may be cause for a refactor PR in the future where numerous blocks could be improved. We are largely hoping to throw away this code by as early as 7.3 so not here but perhaps elsewhere.

Reply:
Ohh, I figured there must be a reason for the lodash syntax, that makes sense now. 

As for idx, there is a README here: https://github.com/elastic/kibana/tree/master/packages/kbn-elastic-idx

For this case I suppose it would be `idx(filter, _ => _.match['monitor.status'])`, which preserves the type better than lodash get. Happy to explain more about it any time. :)

Reply:
@jasonrhodes I think I've read that you can alternatively specify something like:

```typescript
get<any>(foo, `match.${'monitor.status'}`, undefined)
```

But I think the array is better from a readability standpoint.